src/wasi.rs

```diff
@@ -24,10 +24,16 @@ pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
     // https://docs.rs/wasi/0.11.0+wasi-snapshot-preview1/src/wasi/lib_generated.rs.html#2046-2062
     // Note that size of an allocated object can not be bigger than isize::MAX bytes.
     // WASI 0.1 supports only 32-bit WASM, so casting length to `i32` is safe.
+    #[allow(clippy::cast_possible_truncation, clippy::cast_possible_wrap)]
```

Comment:
This is the only place which needs to disable the lints. Unfortunately, the compiler does not use the requirement that non-ZST allocations can not be bigger than `isize::MAX`, so using `TryInto` for converting length into `i32` will contain a panic. We also could return `Error::UNEXPECTED` if `dest.len() > i32::MAX`, but I don't like such workaround.